src/structures/Guild.js

```diff
@@ -942,10 +942,7 @@ class Guild {
         guild_id: this.id,
         role: r,
       });
-      if (data.position) {
-        role.setPosition(data.position, reason);
-        role.position = data.position;
-      }
+      if (data.position) return role.setPosition(data.position, reason);
       return role;
```

Comment:
Could you elaborate? It's being handled inside of the resolve callback, so it should effectively return the role if a position is provided, which is why we have promises in the first place, no?
If the position is not provided, it'll return the role, all goes as expected.